Library/Homebrew/hardware.rb

```diff
@@ -41,6 +41,16 @@ def optimization_flags
       end
       alias generic_optimization_flags optimization_flags
 
+      # Only give values where it is an improvement over rust cpu defaults
+      # Rust already defaults to the oldest supported cpu for each target-triple
+      # Including apple-m1 since Rust 1.71 for aarch64-apple-darwin.
+      def rust_optimisation_flags
+        @rust_optimisation_flags ||= {
+          native:  "-Ctarget-cpu=native",
+          nehalem: "-Ctarget-cpu=nehalem",
```

Comment:
```suggestion
          native:  "--codegen target-cpu=native",
          nehalem: "--codegen target-cpu=nehalem",
```
Please use the long flags for better readability.